src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size + 1 <= elements.length) {
+            elements[size] = value;
+        } else {
+            Object[] moreElements = new Object[(int) (elements.length * 1.5)];
+            System.arraycopy(elements, 0, moreElements, 0, elements.length);
+            moreElements[size] = value;
+            elements = moreElements;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index in method add()");
+        }
+        if (size + 1 <= elements.length) {
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+            elements[index] = value;
+        } else {
+            Object[] moreElements = new Object[(int) (elements.length * 1.5)];
+            System.arraycopy(elements, 0, moreElements, 0, elements.length);
+            elements = moreElements;
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+            elements[index] = value;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index in method add()");
+        }
```

Comment:
you have code like this in get, set, remove. Don’t create [repeating code](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html).